### PR TITLE
Extend description in setup.py

### PR DESCRIPTION
**Pull request by janden**
Right now, our project description on PyPI is [quite terse](https://pypi.org/project/cufinufft/). This PR extends the description a little bit with helpful links, although there is more work to do here.

Also fixes up the author list, URLs, classifiers, and minimum Python version requirements.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,6 +5,13 @@
 
 from setuptools import setup, Extension
 
+# Description
+DESCRIPTION = "Non-uniform fast Fourier transforms on the GPU"
+
+with open('python/cufinufft/README.md', encoding='utf8') as fh:
+    README = fh.read().split('\n')
```
Comment (garrettwrong) [inline setup.py:12]: fh.readlines()

File: setup.py [hunk 2]
```diff
@@ -5,6 +5,13 @@
...
+    LONG_DESCRIPTION = '\n'.join([x for x in README if not x[:3] == '[!['])
```
Comment (garrettwrong) [inline setup.py:13]: Are you trying to match the string literal `'[!['` ?  I didn't see any matches.

**Review by garrettwrong — CHANGES_REQUESTED** [review]
I'm not sure about the readme line filter, otherwise lgtm.

Also if you want these changes included to PyPI registry I think need to retag the code, regen and retest wheels. If you want to see on test pypi first you may have to bump version.

File: setup.py (hunk 2 above)
Comment (janden) [reply setup.py:13]: Copy paste from another project. We don't need it here so will remove.

Comment (garrettwrong) [reply setup.py]: Gotcha, I thought maybe you wanted a regexp for the links or something.

**Comment by janden** [comment]
> Also if you want these changes included to PyPI registry I think need to retag the code, regen and retest wheels. If you want to see on test pypi first you may have to bump version.

Right I will regenerate the wheels and see how they look on Test PyPI (may have to bump version temporarily as you say). Just wanted a second pair of eyes on the new setup code before I started the process. This shouldn't change the core functionality, but if you have time to test it, it's always good to be on the safe side.

**Review by garrettwrong — APPROVED** [review]

**Comment by janden** [comment]
Thanks. Will regenerate the 2010 wheels now.

**Comment by ahbarnett** [comment]
thanks!

**Pull request merged**